utils/upload.go

```diff
@@ -72,6 +75,38 @@ func UploadToCloudStorage(uploader Uploader, path string) {
 	uploader.Upload(walker)
 }
 
+func GetPreSignedURL(videoId string) string {
+	cloudSession := configs.GetCloudSession()
+
+	containerName := constants.CLOUD_CONTAINER_NAME_TEMP
```

Comment:
It would be better if we can Extract the logic for getting the presigned URL into separate functions for each cloud provider, and call the appropriate function based on the cloud provider. For example, you can have a function `getPresignedGCPURL()` that takes the necessary parameters and returns the presigned URL for GCP, and similarly for AWS and Azure. This way, the code for each cloud provider is separated and more readable.

We can move these cloud specific function to a different file as well for code modularity and readability.

```go
func getPresignedGCPURL(videoId string, containerName string) string {
	// Code to generate the presigned URL for GCP
}

func getPresignedAWSURL(videoId string, containerName string) string {
	// Code to generate the presigned URL for AWS
}

func getPresignedAzureURL(videoId string, containerName string) string {
	// Code to generate the presigned URL for Azure
}

func GetPreSignedURL(videoId string, containerName string) string {
	cloudSession := configs.GetCloudSession()

	switch {
	case configs.EnvVar[configs.GCP_PROJECT_ID] != "":
		return getPresignedGCPURL(videoId, containerName)
	case configs.EnvVar[configs.AWS_ACCESS_KEY_ID] != "":
		return getPresignedAWSURL(videoId, containerName)
	case configs.EnvVar[configs.AZURE_ACCESS_KEY] != "":
		return getPresignedAzureURL(videoId, containerName)
	default:
		return ""
	}
}

```

Reply:
yes, was trying to do the same and wanted to share the sessions to each of functions, got this was only, will try to improve this in next fix